services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/endpoints/S3AccessPointEndpointResolver.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.s3.internal.endpoints;
+
+import static software.amazon.awssdk.services.s3.internal.endpoints.S3EndpointUtils.isAccelerateEnabled;
+import static software.amazon.awssdk.services.s3.internal.endpoints.S3EndpointUtils.isArnRegionEnabled;
+import static software.amazon.awssdk.services.s3.internal.endpoints.S3EndpointUtils.isDualstackEnabled;
+import static software.amazon.awssdk.services.s3.internal.endpoints.S3EndpointUtils.isFipsRegion;
+import static software.amazon.awssdk.services.s3.internal.endpoints.S3EndpointUtils.isFipsRegionProvided;
+import static software.amazon.awssdk.services.s3.internal.endpoints.S3EndpointUtils.isPathStyleAccessEnabled;
+import static software.amazon.awssdk.services.s3.internal.endpoints.S3EndpointUtils.removeFipsIfNeeded;
+
+import java.net.URI;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.arns.Arn;
+import software.amazon.awssdk.http.SdkHttpRequest;
+import software.amazon.awssdk.regions.PartitionMetadata;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Configuration;
+import software.amazon.awssdk.services.s3.internal.ConfiguredS3SdkHttpRequest;
+import software.amazon.awssdk.services.s3.internal.resource.S3AccessPointBuilder;
+import software.amazon.awssdk.services.s3.internal.resource.S3AccessPointResource;
+import software.amazon.awssdk.services.s3.internal.resource.S3ArnConverter;
+import software.amazon.awssdk.services.s3.internal.resource.S3OutpostAccessPointBuilder;
+import software.amazon.awssdk.services.s3.internal.resource.S3OutpostResource;
+import software.amazon.awssdk.services.s3.internal.resource.S3Resource;
+import software.amazon.awssdk.services.s3.internal.resource.S3ResourceType;
+import software.amazon.awssdk.utils.Validate;
+
+/**
+ * Returns a new configured HTTP request with a resolved access point endpoint and signing overrides.
+ */
+@SdkInternalApi
+public final class S3AccessPointEndpointResolver implements S3EndpointResolver {
+
+    private static final String S3_OUTPOSTS_NAME = "s3-outposts";
+
+    public static S3AccessPointEndpointResolver create() {
+        return new S3AccessPointEndpointResolver();
+    }
```

Comment:
Usually when we go to the trouble to do this (which we don't tend to for internal classes), there's also a private default constructor.

Reply:
ok!